lib/activerecord_hoarder/batch_archiver.rb

```diff
@@ -1,13 +1,17 @@
 class ::ActiverecordHoarder::BatchArchiver
   def initialize(model_class, storage = nil)
-    @record_collector = ::ActiverecordHoarder::RecordCollector.new(model_class)
+    @batch_collector = ::ActiverecordHoarder::BatchCollector.new(model_class)
     @archive_storage = storage || default_storage_for_records(model_class.table_name)
   end
 
   def archive_batch
-    @record_collector.in_batches(delete_on_success: true) do |batch|
-      success = @archive_storage.store_data(batch)
-      return if !success
+    while @batch_collector.next?
```

Comment:
weird syntax, seems more natural to just do `@batch_collector.each` and not have to worry about calling next? and next_valid

Reply:
and leave choice for validation to configuration/initialization? (So, rather than `each` and `each_valid`, always validate if validation configured)